sdk/resources/mgmt/src/main/java/com/azure/management/resources/fluentcore/profile/AzureProfile.java

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.management.resources.fluentcore.profile;
+
+import com.azure.core.management.AzureEnvironment;
+import com.azure.core.util.Configuration;
+
+/**
+ * Azure profile for client.
+ */
+public class AzureProfile {
+
+    private String tenantId;
+    private String subscriptionId;
+    private AzureEnvironment environment;
+    private final Configuration configuration = Configuration.getGlobalConfiguration();
+
+    /**
+     * Creates AzureProfile instance with Azure environment.
+     *
+     * @param environment the Azure environment
+     */
+    public AzureProfile(AzureEnvironment environment) {
```

Comment:
Maybe improve to either 1. document in javadoc that AzureEnvironment.AZURE can be used as azure global; 2. or an overloaded constructor

Prefer `Objects.requireNonNull`

Reply:
Agree, I will prefer javadoc.